hikari/utilities/net.py

```diff
@@ -39,18 +39,24 @@ async def generate_error_response(response: aiohttp.ClientResponse) -> errors.HT
     real_url = str(response.real_url)
     raw_body = await response.read()
 
+    cls: typing.Optional[typing.Type[errors.HTTPError]] = None
+
     if response.status == http.HTTPStatus.BAD_REQUEST:
-        return errors.BadRequestError(real_url, response.headers, raw_body)
-    if response.status == http.HTTPStatus.UNAUTHORIZED:
-        return errors.UnauthorizedError(real_url, response.headers, raw_body)
-    if response.status == http.HTTPStatus.FORBIDDEN:
-        return errors.ForbiddenError(real_url, response.headers, raw_body)
-    if response.status == http.HTTPStatus.NOT_FOUND:
-        return errors.NotFoundError(real_url, response.headers, raw_body)
+        cls = errors.BadRequestError
+    elif response.status == http.HTTPStatus.UNAUTHORIZED:
+        cls = errors.UnauthorizedError
+    elif response.status == http.HTTPStatus.FORBIDDEN:
+        cls = errors.ForbiddenError
+    elif response.status == http.HTTPStatus.NOT_FOUND:
+        cls = errors.NotFoundError
+
+    if cls is not None:
+        json_body = await response.json()
```

Comment:
Will this be reached and error out if we hit a cloudflare error? (might be worth it defaulting the expected json body fields to "Unknown"

Reply:
This wont get reached since a cloudfare error will not return any of the status there is above

Reply:
Well, you're also covering 400 here which in-theory shouldn't ever be hit if we have enough checks on our end but in rare cases you can get a 400 from services in-between us and and Discord. I can think of two cases of this off the top of my head, when you try to include a payload in a GET request (google will response here) and If you made a request that was too large (I'm sure you get this if the url is too long and that'd likely be caused by including too many elements in a query string but haven't tested to see how this behaves with payload size).